pkg/tempo/automaticloggingprocessor/automaticloggingprocessor.go

```diff
@@ -134,6 +145,21 @@ func (p *automaticLoggingProcessor) ConsumeTraces(ctx context.Context, td pdata.
 	return p.nextConsumer.ConsumeTraces(ctx, td)
 }
 
+func (p *automaticLoggingProcessor) spanLabels(keyValues []interface{}) model.LabelSet {
+	if len(keyValues) == 0 {
+		return nil
+	}
+	ls := make(map[model.LabelName]model.LabelValue, len(keyValues))
+	for i := 0; i < len(keyValues); i += 2 {
+		k := keyValues[i]
+		v := keyValues[i+1]
+		if p.labels[k.(string)] {
+			ls[model.LabelName(k.(string))] = model.LabelValue(v.(string))
```

Comment:
if `k.(string)` is guaranteed to not panic can we just take a `[]string` instead?

Reply:
Changed the logic a bit to make it cleaner and safer.